Vignette

f-lukas opened the pull request:
This is my first draft of the vignette. @arunge or @jan-abel-inwt any tips about what is missing? Thanks!

f-lukas committed:
set up vignette

f-lukas committed:
first draft of vignette

f-lukas committed:
install qpdf

f-lukas commented:
Ich habe gerade noch die Installation von `qpdf` zum Dockerfile hinzugefügt. Jetzt bekomme ich noch die Meldung:

> Pandoc is required to build R Markdown vignettes but not available. Please make sure it is installed.

Gab es diese Probleme auch bei anderen Vignetten @jan-abel-inwt? Soll ich das einfach auch noch im Dockerfile ergänzen?

jan-abel-inwt commented:
Pandoc wird für die markdown Generierung aus deiner Rmd-Vignette benötigt. Die anderen "Apps" mit Vignette haben keine Docker Container Varianten. Du kannst Pandoc hier gerne installieren.

f-lukas committed:
add pandoc installation

f-lukas committed:
adjust dockerfile

f-lukas committed:
fix typo

f-lukas commented:
@jan-abel-inwt danke für die schnelle Rückmeldung. Ich habe es jetzt zum laufen bekommen.

arunge commented on vignettes/how-to-use-MapR.Rmd:
```diff
@@ -0,0 +1,38 @@
+
+## Introduction
+
+To work with MapR, a zip file is needed, which can be uploaded after starting the app. It is also possible to select an online zip files that contains example data. The zip file must contain both the images to be displayed (stored in a folder called `data`) and a `json` file that specifies which images are displayed depending on the users selection.
```
```suggestion
To work with MapR, a zip file is needed, which can be uploaded after starting the app. It is also possible to select an online zip file that contains example data. The zip file must contain both the images to be displayed (stored in a folder called `data`) and a `json` file that specifies which images are displayed depending on the users selection.
```

arunge commented on vignettes/how-to-use-MapR.Rmd:
```diff
@@ -0,0 +1,38 @@
+
+The `json` file can be an image list (`image_list.json`) that contains group-, variable- and measure names and the image belonging to each combination of these three. Also for each image, the path where the image can be found in the `data` folder, must be included.
+
+Alternatively, the `json` file can be a questionnaire (`questionnaire.json`). The file must contain questions that are displayed to the user. Furthermore the file must contain the images to be displayed with the information about for which answer the specific image is displayed. Also the path where the image can be found in the `data` folder, must be included again.
```
```suggestion
Alternatively, the `json` file can be a questionnaire (`questionnaire.json`). The file must contain questions that are displayed to the user. Furthermore the file must contain the images to be displayed with the information which answer is needed to display a specific image. Also the path where the image can be found in the `data` folder, must be included again.
```

arunge commented on README.md:
```diff
@@ -9,3 +9,7 @@ An App to display temporal and temperature graphical files for Isomemo.
+
+## How to use this Package
+
+Refer to the vignette for a description of the usage of the MapR package.
```
```suggestion
Refer to the vignette for a description of the usage of the MapR package.

----

### Notes for developers

When adding information to _help_ sites or the _vignette_ of this package, please update the static HTML documentation with `devtools::build_site()`, and push all changes.
```

arunge replied:
- [ ] We need to add the link to the HTML documentation to the readMe, e.g. just put the link of the HTML vignette under the word "vignette" above. (as soon it is available online.)

jan-abel-inwt replied:
I added this task to the new [Documentaion Issue](https://github.com/Pandora-IsoMemo/MapR/issues/18). I will start with the automation of the doc as soon as this branch is merged into the main branch.

jan-abel-inwt replied:
Sorry I have to reopen this, because it is not quite right, once we automate the documentation workflow. The workflow will always rebuild the documention, when there is change on the main branch. The workflow below is only for locally building the doc.

```suggestion
Refer to the vignette for a description of the usage of the MapR package.

----

### Notes for developers

When adding information to _help_ sites, _docstrings_ or the _vignette_ of this package, 
please update documentation locally as follows:

devtools::document() # or CTRL + SHIFT + D in RStudio
devtools::build_site()
```

arunge reviewed (changes requested):
As discussed, and please see the minor remarks for the ReadMe and the vignette.

f-lukas committed:
Update vignettes/how-to-use-MapR.Rmd

Co-authored-by: Antonia Runge <antonia.runge@inwt-statistics.de>

f-lukas committed:
Update vignettes/how-to-use-MapR.Rmd

Co-authored-by: Antonia Runge <antonia.runge@inwt-statistics.de>

f-lukas committed:
Update README.md

Co-authored-by: Antonia Runge <antonia.runge@inwt-statistics.de>

jan-abel-inwt commented:
I think we should link the documentation in the app itself. Maybe Somewhere here (Screenshot of the iso-app):

![image](https://github.com/Pandora-IsoMemo/MapR/assets/106665518/1e4cd9b1-7fce-4e03-8b9f-31fd86669f22)


Should this be done via data-tools @arunge ?

jan-abel-inwt commented:
@f-lukas the link to the main version is missing in the Readme

f-lukas committed:
update vignette and readme

f-lukas commented:
> @f-lukas the link to the main version is missing in the Readme

Thanks, I added it.

I also added what we discussed in the vignette @arunge.

> I think we should link the documentation in the app itself. Maybe Somewhere here (Screenshot of the iso-app):

I will add this in another branch. So you can already start with the automation once I merged this branch @jan-abel-inwt. In parallel I will add the header buttons as part of shiny-tools.

f-lukas committed:
Update README.md

Co-authored-by: Jan Abel <106665518+jan-abel-inwt@users.noreply.github.com>

arunge reviewed (changes requested):
Very nice description, please check the formatting [here](https://github.com/Pandora-IsoMemo/MapR/pull/19).

arunge committed:
adding little structure into the vignette (#19)

f-lukas committed:
small readme fix
